File: src/CSharp/91.DecodeWays.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.NumDecodings
{
public class Solution {
    public int NumDecodings(string s) {
        int pp = 1;
        int p = (s[0] != '0') ? 1 : 0;
        for (int i = 1; i < s.Length; i++) {
            int twonumber = (s[i - 1] - '0') * 10 + s[i] - '0';
            int onenumber = s[i] - '0';
            int temp = p;
            p = pp * ((twonumber > 9 && twonumber <= 26) ? 1 : 0) + p * (onenumber > 0 ? 1 : 0);
            pp = temp;
        }
        return p;
    }
}
}

File: src/Python/91.DecodeWays.py

class Solution:
    def numDecodings(self, s: str) -> int:
        pp, p = 1, int(s[0] != '0')
        for i in range(1, len(s)):
            pp, p = p, pp * (9 < int(s[i-1:i+1]) <= 26) + p * (int(s[i]) > 0)
        return p

if __name__ == "__main__":
    solution = Solution()
    print(solution.numDecodings("12"))
    print(solution.numDecodings("226"))
    print(solution.numDecodings("266"))
    print(solution.numDecodings("2224"))
    print(solution.numDecodings("1010"))
